javascript-review.js

```diff
@@ -22,18 +22,35 @@
 
 // What is JSON? Be clear but concise.
 
+const json = "JavaScript Object Notation is a condesed javascript based storage file-format that stores data in objects (hashes). Javascript has a variety of methods specifically relating to making and reading json files" 
+
 // 6.
 
 // What is a callback?
 
+const callback = "A callback is when a function is called as if it was an annonymous function from inside some other javascript block."
+
 // 7.
 
 // Define a function with that takes two number arguments and a callback. The function will add the two numbers and pass them to the callback as an argument. Now call this function and in the callback simply console.log out the argument that has been passed through.
 
+const print = (arg) => {
+    console.log(arg)
+}
+
+const myFunction = (num1, num2, callback) => {
+    const answer = num1 + num2;
+    callback(answer)
+}
+
+myFunction(1,2,print)
+
 // 8.
 
 // Why do we use promises?
 
+
+
 // 9.
 
 // What are the two methods at our disposal if we have hold of a promise? What are the names of the functions that are related to these functions (that are called when the promise is fulfilled or fails).
```

Comment:
remember that we have `.then` and `.catch`, we have `resolve` and `reject` when we're defining our own promise